src/pages/FundingOpen/GiftSetPage.jsx

```diff
@@ -1,32 +1,64 @@
 import styled from 'styled-components';
-import { useState } from 'react';
+import { useState, useContext } from 'react';
+import { DataContext, PageContext } from './IndexPage';
 import BackHeaderComponent from '../../components/common/BackHeaderComponent';
 import PriceInputComponent from '../../components/common/PriceInputComponent';
 import BottomBackgroundComponent from '../../components/common/BottomBackgroundComponent';
 import ButtonComponent from '../../components/common/ButtonComponent';
 import icn_plus from '../../assets/FungingOpen/icn_plus.svg';
 
-const GiftSetPage = () => {
-  const [price, setPrice] = useState(null);
-  const [url, setUrl] = useState(null);
-  const [imageFile, setImageFile] = useState(null);
-  const [imagePreview, setImagePreview] = useState(null);
-
-  const handleUrlChange = (event) => {
-    setUrl(event.target.value);
-  };
-  const handleImageChange = (event) => {
-    const file = event.target.files[0];
-    if (file) {
-      setImageFile(file);
-      setImagePreview(URL.createObjectURL(file));
+const GiftSetPage = ({
+  lastGiftData = {
+    giftName: null,
+    price: null,
+    giftUrl: null,
+  },
+  lastImageData = null,
+}) => {
+  const { setCurrentPage } = useContext(PageContext);
+  const { setGiftData, setImageData } = useContext(DataContext);
+
+  const [name, setName] = useState(lastGiftData.giftName);
+  const [price, setPrice] = useState(lastGiftData.price);
+  const [url, setUrl] = useState(lastGiftData.giftUrl);
+  const [imageFile, setImageFile] = useState(lastImageData);
+
+  const handleFormSubmit = () => {
+    if (lastGiftData) {
+      setGiftData((prevItems) => [
+        // 마지막 아이템 삭제
+        ...prevItems.slice(prevItems.length - 2),
```

Comment:
```suggestion
        ...prevItems.slice(0, -1),
```
이렇게 잘라주거나 
```suggestion
        ...prevItems.slice(0, prevItems.length - 1),
```
요렇게 잘라줘야 할 듯 합니다!